extensions/sockets/src/main/java/org/mule/extension/socket/api/provider/udp/UdpRequesterProvider.java

```diff
@@ -35,21 +37,21 @@
   /**
    * This configuration parameter refers to the address where the {@link DatagramSocket} should send packages to.
    */
-  @ParameterGroup
-  private ConnectionSettings connectionSettings;
+  @ParameterGroup(CONNECTION)
+  private SocketConnectionSettings connectionSettings;
 
   /**
    * {@link DatagramSocket} configuration properties
    */
-  @ParameterGroup
+  @ParameterGroup(SocketProperties.GROUP_NAME)
   private UdpSocketProperties udpSocketProperties;
 
   /**
    * This configuration parameter refers to the address where the {@link DatagramSocket} should bind to.
    */
   @Parameter
   @Optional
-  private ConnectionSettings localAddressSettings = new ConnectionSettings();
+  private SocketConnectionSettings localAddressSettings = new SocketConnectionSettings();
```

Comment:
org.mule.runtime.extension.api.exception.IllegalParameterModelDefinitionException: Class 'org.mule.extension.socket.api.SocketConnectionSettings' cannot be used with '@NullSafe' parameter since it contains non optional fields